reconcile/openshift_saas_deploy_trigger_base.py

```diff
@@ -63,12 +67,158 @@ def setup(options):
     return saasherder, jenkins_map, oc_map, settings
 
 
-def construct_tekton_trigger_resource(saas_file_name,
-                                      env_name,
-                                      timeout,
-                                      settings,
-                                      integration,
-                                      integration_version):
+def trigger(spec,
+            dry_run,
+            jenkins_map,
+            oc_map,
+            already_triggered,
+            settings,
+            state_update_method,
+            integration,
+            integration_version):
+    """Trigger a deployment according to the specified pipelines provider
+
+    Args:
+        spec (dict): A trigger spec as created by saasherder
+        dry_run (bool): Is this a dry run
+        jenkins_map (dict): Instance names with JenkinsApi instances
+        oc_map (OC_Map): a dictionary of OC clients per cluster
+        already_triggered (set): A set of already triggered deployments.
+                                    It will get populated by this function.
+        settings (dict): App-interface settings
+        state_update_method (function): A method to call to update state
+        integration (string): Name of calling integration
+        integration_version (string): Version of calling integration
+
+    Returns:
+        bool: True if there was an error, False otherwise
+    """
+
+    # TODO: Convert these into a dataclass.
+    saas_file_name = spec['saas_file_name']
+    provider_name = spec['pipelines_provider']['provider']
+
+    error = False
+    if provider_name == 'jenkins':
```

Comment:
Can we have an `Enum` with the providers?

In some file:

```python
import enum
class Providers(enum.Enum):
    JENKINS = "jenkins"
    TEKTON = "tekton"
```
and here it becomes:
```python
if provider_name == Providers.JENKINS.value:
    ...
```

Or at least have some named constants. My concern is that we will eventually evolve this and somebody will type `jeenkins` here.